src/mlpack/methods/ann/loss_functions/poisson_nll_loss.hpp

```diff
@@ -0,0 +1,128 @@
+/**
+ * @file poisson_nll_loss.hpp
+ * @author Mrityunjay Tripathi
+ *
+ * Definition of the Poisson NegativeLogLikelihood class.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_POISSON_NLL_LOSS_HPP
+#define MLPACK_METHODS_ANN_LAYER_POISSON_NLL_LOSS_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * Implementation of the poisson negative log likelihood layer. The poisson 
+ * negative log likelihood layer expects input for each class. 
+ * The layer also expects a class index, in the range between 1 and the
+ * number of classes, as target when calling the Forward function.
+ *
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat
+>
+class PoissonNLLLoss
+{
+ public:
+  /**
+   * Create the PoissonNLL object.
+   * 
+   * @param log_input if true the loss is computed as exp(input) - target*input, if
```

Comment:
The description should start with a capital letter.

Reply:
Ok.